### ISSUE TITLE
Background/foreground rendering strangely for certain unicode characters

### ISSUE THREAD

**Issue by noprobelm** [asciinema/agg#63]
**Describe the bug**
Hi, thanks for creating such an awesome tool.

I apologize in advance if this is a user error, but I can't seem to figure it out. When recording a ▄ character in the terminal, the background/foreground render unexpectedly and unpredictably (there seems to be some kind of overlap in the colors where there shouldn't be). 

The corresponding `.cast` playback looks as I would expect it to. Here are some examples:

![conway](https://github.com/asciinema/agg/assets/7587293/4b97465e-cf8e-4664-9e5e-9337cd2522fa)

![example](https://github.com/asciinema/agg/assets/7587293/1add51ff-b3dd-421f-bead-98ac40c7f391)

**To Reproduce**
1. Record a ▄ character in the termianl (ensure the background and foreground are different colors). Convert it to a gif using `agg`

**Expected behavior**
No background/foreground overlap should occur

**Versions:**
 - OS: EndeavourOS
 - agg 1.4.3

Output with verbose flag:
```
[INFO  agg] terminal size: 140x35
[INFO  agg] selected font families: ["JetBrains Mono", "Fira Code", "DejaVu Sans Mono", "Liberation Mono", "DejaVu Sans"]
[INFO  agg] selected theme: dracula
[INFO  agg] gif dimensions: 1193x706
```

**Comment by ku1ik** [comment]
Could you paste the cast file for the 2nd gif? It would help for me to know what characters and control sequences were recorded in the first place. Thanks.

**Comment by noprobelm** [comment]
I discarded the old cast file, here's a new one I just recorded. It uses the same character. It's worth noting this will happen with other [block element](https://www.compart.com/en/unicode/block/U+2580) characters as well.

[lower_half_block.txt](https://github.com/asciinema/agg/files/14312435/lower_half_block.txt)

**Comment by ku1ik** [comment]
This has been fixed in the recently released agg 1.5.0: https://github.com/asciinema/agg/releases/tag/v1.5.0

![lhb](https://github.com/user-attachments/assets/a3eacf36-7dd1-4ee4-b89a-b5b854ad3ed8)
